### PR TITLE
added replace to string

### PR DESCRIPTION
**Pull request by bjartek**
## Description

Add replace(from:String, to:String) to String.

______

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
Thank you for contributing this @bjartek!

One open question is what the behaviour in regards to equivalent Unicode sequences should be. Cadence strings are encoded as UTF-8 and logically consist of Unicode grapheme clusters, like in Swift, but unlike Go, where strings are just byte arrays.

This also means that strings replacements should operate on logical characters, i.e. Unicode grapheme clusters, and not purely on bytes.

For example, these two strings are equivalent: `"Caf\u{65}\u{301}" == "Caf\u{e9}"`.

However, observe the result in Go:

```go
strings.ReplaceAll("Caf\u0065\u0301", "\u00E9", "X") == "Caf\u0065\u0301"
```

Go's `strings.ReplaceAll` operates on bytes, and the searched byte sequence does not occur in the source.

Now compare this to Swift:

```swift
"Caf\u{65}\u{301}".replacingOccurrences(of: "\u{E9}", with: "X") == "CafX"
```

I don't know of the top of my head what algorithm is used for the replacement, it could be that Unicode normalization is involved, like when testing equality.

**Comment by bjartek** [comment]
Good questions. I have not though that far tbh. Maybe we can look at a different package to do the replacement in go. Or call the method replaceAscii or something?

**Comment by bjartek** [comment]
this is over a. year old @turbolent should we just close it?

File: runtime/sema/string_type.go [hunk 1]
```diff
@@ -195,3 +206,23 @@ var StringTypeToLowerFunctionType = &FunctionType{
...
 `
+
+var StringTypeReplaceFunctionType = &FunctionType{
+	Parameters: []*Parameter{
+		{
+			Identifier:     "from",
+			TypeAnnotation: NewTypeAnnotation(StringType),
+		},
+		{
+			Identifier:     "upTo",
```
Comment (bluesign) [inline runtime/sema/string_type.go:217]: s/upTo/to

**Comment by bluesign** [comment]
oh I thought this is new PR, this would be very nice to have.

**Comment by turbolent** [comment]
Yes, if you want to. We can always re-open it

**Pull request closed without merging**